### PR TITLE
OC API for gtce and gregicality

### PR DESCRIPTION
**Pull request by Yefancy**
I really like **Gregicality's** work and ideas, and I'd like to do some stuff for **Gregicality**. As I expected on Discord yesterday, I have implemented some API functions. It may not cover all interactions, but I believe it covers most of the useful ones. The changes are as follows:
- `IWorkable`
- `IEnergyContainer`
- `MultiblockRecipeLogic`
- `AbstractRecipeLogic`
- `SimpleMachineMetaTileEntity`
- `CoverBehavior`

Basically, it contains obtaining simple/mutil machines' status, input, output, and currentRecipe. 
Supporting interaction with cover, autoOutputSetting , and working status.

In fact, I don't know enough about features of all the multi-blocks, only interacting with some of the underlying classes. Some APIs for multi-block may be added in the future (e.g. `overheat` of void Miner or something like that). 

I was also considering to add the `Computer Block` as a building material for multi-block to solve the problem that many multi-block have only one face to connect an adapter. Of course, because of the complexity of the implementation, an unnecessary potential feature needs to be discussed.

### CODE REVIEW

File: src/main/java/gregicadditions/integrations/opencomputers/driver/environment/EnvironmentMetaTileEntity.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package gregicadditions.integrations.opencomputers.driver.environment;
+
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import li.cil.oc.api.driver.NamedBlock;
+import li.cil.oc.integration.ManagedTileEntityEnvironment;
+
+public abstract class EnvironmentMetaTileEntity<T> extends ManagedTileEntityEnvironment<T> implements NamedBlock {
+    private String preferredName = "gtce_workable";
```
Comment (huneau) [inline src/main/java/gregicadditions/integrations/opencomputers/driver/environment/EnvironmentMetaTileEntity.java:8]: gtce_workable ?

**Review by huneau — CHANGES_REQUESTED** [review]
nice job

**Comment by Yefancy** [comment]
Please excuse my oversight. While they don't affect functionality, this really shouldn't happen.

**Comment by huneau** [comment]
perfect job, can you send me some screenshot of the result ?

**Comment by Yefancy** [comment]
image
machines proxy.

image
muti-block example

image
few fluid in large chemical reactor

image
obtain bus or hatch.

image
image
machine's status

image
image
getCurrentRecipe

image
image
cover

image
image
cover config

image
image
machine autoOutput

Thank you for waiting. There are so many functions involved in the implementation that I can't show them all. I selected some of features to make screenshots.

**Comment by huneau** [comment]
perfect, I will make it for 0.21

**Pull request merged**